api/commentRouter.js

```diff
@@ -0,0 +1,59 @@
+const express = require("express");
+const Comment = require("../models/commentModel");
+const commentRouter = express.Router();
+
+commentRouter.route("/:cityId").get((req, res) => {
+  Comment.find({ city: req.params.cityId })
+    .populate("city", "name")
+    .populate("itinerary", "title")
+    .exec((err, comments) => {
+      //we send the comments sorted by date, so the newest ones goes on top
+      res.json(comments.sort((a, b) => a.date < b.date));
+    });
+});
+
+commentRouter
+  .route("/find/:id")
+  .get((req, res) => {
+    Comment.find({ _id: req.params.id }, (err, comment) => {
+      res.json(comment);
+    });
+  })
+  .delete((req, res) => {
+    Comment.findById(req.params.id, (err, comment) => {
+      if (
+        comment.user.name ===
```

Comment:
For readability, try to avoid nesting ternary's inside if statements.  This would read a lot easier if you set a variable before and then referenced it within the if.